dan_layer/consensus/src/hotstuff/on_receive_proposal.rs

```diff
@@ -133,6 +133,15 @@ where TConsensusSpec: ConsensusSpec
         local_committee: Committee<TConsensusSpec::Addr>,
         block: Block<TConsensusSpec::Addr>,
     ) -> Result<(), HotStuffError> {
+        if self.store.with_read_tx(|tx| block.get_parent(tx)).is_err() {
```

Comment:
Same as above

```suggestion
        if self.store.with_read_tx(|tx| block.get_parent(tx)).optional().is_none() {
```

We should make Block::exists a non-instance function rather:

```rust
    pub fn exists<TTx: StateStoreReadTransaction<Addr = TAddr> + ?Sized>(
     
        tx: &mut TTx,
block_id: &BlockId    
) -> Result<bool, StorageError> {
        tx.blocks_exists(block_id)
    }
```